src/prefect/agent/docker/agent.py

```diff
@@ -330,11 +338,15 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
 
         if not self.no_pull and len(image.split("/")) > 1:
             self.logger.info("Pulling image {}...".format(image))
-
-            pull_output = self.docker_client.pull(image, stream=True, decode=True)
-            for line in pull_output:
-                self.logger.debug(line)
-            self.logger.info("Successfully pulled image {}...".format(image))
+            registry = image.split("/")[0]
+            if self.reg_allow_list and registry not in self.reg_allow_list:
+                self.logger.warning("Trying to pull image from a Docker registry '{}' which"
+                                    " is not in the reg_allow_list".format(registry))
```

Comment:
Instead of continuing on a non-authorized image I think this should raise a ValueError or something of the like (and also keep the error log). Then the outside call to the `deploy_flows` function will safely trap the error and fail the flow run with this reason

Reply:
Should it be an error instead of warning then?

Reply:
Yeah let's update it to an error log